src/theory/bv/bv_solver_bitblast.cpp

```diff
@@ -26,6 +26,46 @@ namespace cvc5 {
 namespace theory {
 namespace bv {
 
+/**
+ * Notifies the BV solver when assertions were reset.
+ *
+ * This class is notified after every user-context pop and maintains a flag
+ * that indicates whether assertions have been reset. If the user-context level
+ * reaches level 0 it means that the assertions were reset.
+ */
+class NotifyResetAssertions : public context::ContextNotifyObj
+{
+ public:
+  NotifyResetAssertions(context::Context* c)
+      : context::ContextNotifyObj(c, false),
+        d_context(c),
+        d_reset_assertions_notified(false)
+  {
+  }
+
+  bool notifiedResetAssertions() { return d_reset_assertions_notified; }
```

Comment:
Suggestion: it sounds a bit like this method notifies reset-assertions. Maybe `doneAssertionReset`?